tests/core/data/test_network.py

```diff
@@ -352,6 +353,80 @@ def test_from_2d_array(shape, expected_dims):
     assert network.dims == expected_dims
 
 
+@pytest.mark.parametrize(
+    "dim_edges, target_dims",
+    [
+        ([4], [2, 2]),
+        ([8], [2, 2, 2]),
+        ([2, 2], [2, 2]),
+        ([2, 4, 2], [2, 2, 2, 2]),
+        ([2, 15, 7], [2, 3, 5, 7]),
+    ],
+)
+class TestMatchDims:
+    """This class contians all the tests for the functions that change the
+    dimsnension of a network by splitting edges.
+    """
+
+    def test_match_dimensions(self, dim_edges, target_dims):
+        node = random_node(dim_edges)
+        network = Network(node[:], [], copy=False)
+        edges = _match_dimensions(network.out_edges, target_dims)
+        assert [e.dimension for e in edges] == target_dims
+        # We do not test for the edges in `node.edges` being properly ordered.
+        # This is because (surprisingly) split_edges does not repect the order of
+        # edges in a node.
+        # For network operations this will not be a problem.
+        assert set(node.edges) == set(edges)
+
+    def test_match_out_dims(self, dim_edges, target_dims):
+        node = random_node(dim_edges)
+        network = Network(node[:], [], copy=False)
+        new_network = network.match_out_dims(target_dims)
+        assert new_network is network
+        assert network.dims[0] == target_dims
+
+    def test_match_in_dims(self, dim_edges, target_dims):
+        node = random_node(dim_edges)
+        network = Network([], node[:], copy=False)
+        new_network = network.match_in_dims(target_dims)
+        assert new_network is network
+        assert network.dims[1] == target_dims
+
+
+@pytest.mark.parametrize(
+    "dim_edges, target_dims",
+    [
+        ([2, 2], [4]),
+        ([2, 2], [3]),
+        ([2, 3, 5], [2, 5, 3]),
+    ],
+)
+class TestMatchDimsRaisise:
+    """This class contians the tests for the functions that change the
+    dimsnension of a network by splitting edges. In particular it ensures that
+    thep proper error is raised.
+    """
+
+    def test_match_dimensions_raises(self, dim_edges, target_dims):
+        node = random_node(dim_edges)
+        network = Network(node[:], [], copy=False)
+        with pytest.raises(ValueError):
+            _match_dimensions(network.out_edges, target_dims)
+
+    def test_match_out_dims_raises(self, dim_edges, target_dims):
+        node = random_node(dim_edges)
+        network = Network(node[:], [], copy=False)
+        with pytest.raises(ValueError):
+            network.match_out_dims(target_dims)
```

Comment:
It's probably worthwhile asserting that `network` is in a valid state after an error has been raised.

Reply:
That's a good  point. I will do so.